File: runtime/src/test/java/com/siimkinks/sqlitemagic/SqlTestAssertions.java
package com.siimkinks.sqlitemagic;

import java.util.ArrayList;

import static com.google.common.truth.Truth.assertThat;

final class SqlTestAssertions {
  private SqlTestAssertions() {
  }

  static void assertExpression(
      Expr expression,
      String expectedSql,
      String... expectedArgs
  ) {
    final StringBuilder sql = new StringBuilder();
    final ArrayList<String> args = new ArrayList<>();
    expression.appendToSql(sql);
    expression.addArgs(args);

    assertThat(sql.toString()).isEqualTo(expectedSql);
    assertThat(args).containsExactlyElementsIn(expectedArgs).inOrder();
  }
}

File: runtime/src/test/kotlin/com/siimkinks/sqlitemagic/SqlTestAssertions.kt
package com.siimkinks.sqlitemagic

import com.google.common.truth.Truth.assertThat

object SqlTestAssertions {
  fun assertExpression(
    expression: Expr,
    expectedSql: String,
    vararg expectedArgs: String?
  ) {
    val sql = StringBuilder()
    val args = ArrayList<String?>()
    expression.appendToSql(sql)
    expression.addArgs(args)

    assertThat(sql.toString()).isEqualTo(expectedSql)
    assertThat(args).containsExactlyElementsIn(expectedArgs).inOrder()
  }
}
